core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/BatchWritableSink.java

```diff
@@ -49,7 +54,11 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     properties.remove(Properties.BatchReadableWritable.NAME);
     properties.remove(Properties.BatchReadableWritable.TYPE);
 
-    pipelineConfigurer.createDataset(datasetName, datasetType, DatasetProperties.builder().addAll(properties).build());
+    // null check for tests, as macro fields map and properties are null in test cases
```

Comment:
if its just that one test that is failing, I think its enough just to move the test to SchemaValidatorTest.